src/node/tendermint_data.ml

```diff
@@ -0,0 +1,302 @@
+open Crypto
+open State
+
+open Tendermint_helpers
+
+open Tendermint_internals
+(** Tendermint input_log and output_log.
+    Holds all the querying function on the input_log required by Tendermint subprocesses. *)
+
+module CI = Tendermint_internals
+
+type round = CI.round
+
+type height = CI.height
+
+module MySet = Set.Make (struct
+  type t = CI.value * CI.round
+
+  let compare (v1, r1) (v2, r2) = compare (v1, r1) (v2, r2)
+end)
+
+(* FIXME: Tendermint we could do better *)
+type node_identifier = Key_hash.t
+
+type index_step = CI.consensus_step
+
+type index = height * index_step
+(** Tendermint input is indexed by height and consensus step. *)
+
+type proposal_content = {
+  process_round : round;  (** Round of the process that sent the PROPOSAL *)
+  proposal : CI.value;
+  process_valid_round : round;
+      (** "Locked round" (see paper) of the PRECOMMIT of this value; -1 when the value has notbeen locked before *)
```

Comment:
You can use the markup for odoc: 
`@see <url> message`:	add a reference to the URL with a message as a comment